red-service/src/main/java/com/masi/red/TestService.java

```diff
@@ -23,62 +26,58 @@
 public class TestService implements ITestService {
 
     private final TestRepository testRepository;
-    private final JobTitleRepository jobTitleRepository;
-    private final UserRepository userRepository;
+    private final EntityFinder entityFinder;
+    private final MapperFacade mapper;
 
     @Override
-    public Test addTest(NewTestDTO testDTO) {
-        JobTitle jobTitle = jobTitleRepository.findById(testDTO.getJobTitleId())
-                .orElseThrow(() -> new EntityNotFoundException("Nie znaleziono stanowiska o id " + testDTO.getJobTitleId()));
+    public TestDTO addTest(NewTestDTO testDTO) {
+        JobTitle jobTitle = entityFinder.findJobTitleById(testDTO.getJobTitleId());
         Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
```

Comment:
mozesz to wyciagnac w kontrolerze przy użyciu `@AuthenticationPrincipal User user` jako argument w endpoincie, np. 
`@GetMapping("/")
    public String home(@AuthenticationPrincipal User user)`